### ISSUE TITLE
Remote Control

### ISSUE THREAD

**Issue by Chipperchap** [AntonyCorbett/OnlyM#23]

**Comment by AntonyCorbett** [comment]
@Chipperchap there are no definite plans. Instructions are that media is controlled by sound desk, so remote control would be an edge case.

**Comment by Chipperchap** [comment]
Yes, I agree with the instructions for regular congregational use, however I use SoundBox remote for training purposes as well, such as 'Sound, Platform and Attendants', 'sign language', as well as 'LDC' training - so unless I had an assistant to advance the next 'slide' (image or similar) I would have to be sound booth bound.  
For what it is worth, I use images in my public talks, and some congregations I visit are just not set up for image display, despite my giving them a detailed prompt sheet to work to... so it is just easier to control the content from my remote app via my phone rather than train booth personnel just before the meeting... but I confess that bends the instructions a little...

**Comment by Chipperchap** [comment]
Thank you Omaha4.
I always test things first on my home PC before attempting to do anything elsewhere and alas I don't have a wifi card on my desktop.  I know the PC at the K-Hall does though so I could try it... and our sign language laptop may support it as well so I can investigate it.
Most K-Halls I go to use laptops for their media (why would they do that!) so it is possible to use that option I suppose.  I actually don't know anything about the 'project to this PC' option with W10 so I will do my homework.
For what it is worth, when I did some LDC training today my son was a great help dealing with the K-Hall I was at laptop while I did the speaking so it worked out okay <smile>.

**Comment by p-ho** [comment]
To add another view to that discussion:
As a foreign-language group we hold most of our meetings in the auxiliary room of our khall. There we don't have a special "sound desk" and only one screen to operate on.
Consequently, in order to avoid unnecessary distraction we rely on the SoundBox remote functionality for our regular meetings. For us it's not an edge case.
The same should be true for multiple other groups not using the main auditorium of their hall.

**Comment by AntonyCorbett** [comment]
@p-ho Thanks for the feedback. When you move to JW Library you will need a second monitor and local control of playback, which will reduce the requirement for an OnlyM remote to an edge case.

**Comment by Chipperchap** [comment]
Hi p-Ho,
When you say "we... have... only one screen to operate on", how do you currently control the content with SoundBox - is it attached to a laptop...?

We currently have a laptop sitting on a shelf on a mobile trolley with a 40" screen above it for our Sign Language needs.  Our language (NZSL) is not supported by the JWLanguage app (and is looking unlikely as there is relatively little 'need' for it population-wise) (or JWL at this stage) so we will likely be reliant on SoundBox well after the unsupported period starts but that is fine.

I tried the 'Projecting to this screen' as mentioned above but the laptop we have (an older DELL) doesn't have hardware that supports the feature. 

Never mind, I am extremely grateful for what is currently available and use it to the full.

**Comment by omaha4** [comment]
That is a shame, “Project to this screen” didn’t work out. By NZSL, you mean New Zealand Sign Language correct?

Get Outlook for iOS<https://aka.ms/o0ukef>
________________________________

**Comment by omaha4** [comment]
@chipperchap, when you say JWLanguage, do you mean the green app, or the orange app. Im in American Sign Language, here is the reason I’m asking:[Image]


Get Outlook for iOS<https://aka.ms/o0ukef>
________________________________

**Comment by p-ho** [comment]
@Chipperchap 
We use a nettop PC that is directly connected to the main screen. For control a wireless keyboard and SoundBox Media Remote is used.

**Comment by Chipperchap** [comment]
omaha4:
Yes, NZSL = New Zealand Sign Language

I was meaning the green one - then when you mentioned the orange app, I had forgotten about that one so I checked it out.  Wow, it now has NZSL content (albeit limited).  I downloaded to my desktop PC and it projects to a second screen which is interesting.  We still have no way to control it as our laptop is away from our seating (on the trolley) and even using a wireless mouse is not practical.  We tried using a 5m VGA cable in the past with the laptop in the seating area on a TV table (with castors) but it is just too clunky as it is a tripping hazard and evacuation hazard and we have to dismantle after each meeting for the other congregations that share the K-Hall.  Another alternative is to use something like 'Remote Desktop' to control the laptop via a tablet, but it is a pity the JWSL app doesn't have meeting program content like JWL.  Thanks for the reminder and query though. 

Question: In JWSL app, how do you select a scripture (chapter) without it *auto-playing* once it loads, especially if you want to display another scripture other than the first one, that is, another verse in the available list?

**Comment by omaha4** [comment]
@chipperchap : 1) use the playlist feature JWSL, it will become your friend in a hurry. 2) You can use an Apple TV and air play it from an iOS device to an Apple TV, newer Apple products render better results, make sure it’s done over a network though, or 3) Use a long HDMi cable or some SDI to HDMI converters and a length of coax to put your cart where you need it, and tape to the floor using gaffers tape. As for setup/tear down, well, that’s part of it right now. Between using our Apple TVs and projecting to our primary pc from my personal pc when needed, we have got our setup/tear down almost nothing, but investments had to be made, both in KH infrastructure and in personal equipment.


[Image]

Get Outlook for iOS<https://aka.ms/o0ukef>
________________________________

**Comment by Chipperchap** [comment]
1) Thank you for the tip @omaha4 

2) I use Apple TV at home for all our family meeting prep etc. so am thoroughly familiar with the concept, but I have spent so much over the years on our sign language dept. at the K-Hall that I am reluctant to spend yet more resources.  As you say, 'investment' has to be made.  Perhaps one day... 

3) Of course I don't know the configuration of your setup at your K-Hall but taping coax / HDMi cables etc. is just not an option at our hall.  For a start there is a fire exit door in the way and we also have a vaulted ceiling (no suspended ceiling) among a number of other issues, so when I say a *mobile* trolley, that is what I mean, it has to be totally mobile and as wireless as possible which is why I am so keen on a wireless interface control which SoundBox and the SoundBox remote app are currently excellent for.  

Looks like SoundBox will continue to be our go-to option for the foreseeable future... that is until the new world when the "ears of the deaf will be unstopped"! [smile]

Thank you for your comments though, really appreciate it.

**Comment by omaha4** [comment]
Yea, taping to the floor is one of those things you only do "if you have to."  For us, we are a small group, in a small town, with two deaf publishers. But we are defiantly not a fringe case as we have small language groups  (there is people that even speak those languages around here?) poping up like hot cakes within a hour, hour and a half drive in two directions of travel. It has been amazing to see. 

We use a combo of methods. In the second school, we have an apple tv which we stream our talks to and conduct our WT with the JWSL app. For the midweek meeting, we use the main hall and the main monitors. We have a switcher installed that has both manual and ip controls. For the most of the meeting, we stream the parts to the appletv. For the congregation book study, We either project from my Surface to the main PC to the screens, or airplay mirror to the apple tv to the main screens. Wifi and Airplay reliability has been checkered here lately, I may start dropping an SDI line between my Surface and our switcher for the midweek to reduce the glitches.

We have limited personal, so we have had to go extra lengths to pull the meetings off. But as I have been instructed, whatever it takes for our new publishers to benefit and grow is what we need to do. Frankly, while being part of a small language group has its challenges, it has been really exciting to see so many pop up in area.

**Comment by VirtSecurity** [comment]
I see the instruction for not having the brother on stage remote control the software we use it to separate the function of video systems brother and sound systems brother.  We do not have much room at our sound console for two brothers to sit together there.  We have used the remote control on SoundBox to allow the video systems brother to sit in the audience and run the pictures/videos from their seat.  This allows some of our older brothers to have a share in helping with the meeting.  They are some of the most consistent in caring for this assignment and it only took a little bit of training.

Cross posted from another thread here in issues...  :D

Love the system.  Wish it had been adopted and combined with JWL.

**Comment by p-ho** [comment]
@AnsNie 
> Now I have changed my mind though, and I think that while media control should be in the AV-desk as much as possible, it would be nice to have an option that speakers could control their own images.

Let me add as a reminder: There are indeed meetings held at locations where there isn't and really won't be a dedicated AV-desk.

**Comment by AnsNie** [comment]
@p-ho 
> Let me add as a reminder: There are indeed meetings held at locations where there isn't and really won't be a dedicated AV-desk.

You are right, I didn't think of that (also sorry for not reading really carefully all what is written above =D)
I was speaking generally - in most cases public talk images should be the only need of remote control, as everything else comes from JWL (unless brothers manually download media to OnlyM). 
Anyways, knowing that there is also these situations in some congregations, only adds more reasons why OnlyM should have option to use remote control.

The only aim is to get the meetings (and especially meeting media) to run smoothly, making possible for everyone to focus on the spiritual food better, which is most important of course. 
In case of remote control, if users have skills to use it, it would benefit all, as I wrote above. If they don't know how to use it, it won't be any help. The thing is just that I don't believe such person would even try using it, if he barely knows what such feature is. From this view I don't think we need to fear that unskilled brothers will mess things up trying to use the remote feature, as they will be happy using the simple PC user interface anyways.

**Comment by AntonyCorbett** [comment]
@AnsNie We have written instructions that AV is to be controlled at the desk rather than by the speaker. I think this must categorise the feature as an edge case. Agreed?

**Comment by AntonyCorbett** [comment]
@AnsNie thanks for clarification. Yes, it's a difficult call isn't it? I think I have to conclude that in a standard KH meeting setting (our target deployment), the special requirements of a few would be met by remote control of OnlyM, but a large majority of users have no need for it. And although some may see a benefit in RC, they nevertheless choose to follow the instruction received. Of course, OnlyM is an open-source project and others are free to fork the code... Another consideration is that a mobile app would also need to be created - no small task - and in this I am dependent on other devs. 

Thanks as always for your useful feedback. Antony

**Comment by Chipperchap** [comment]
At the risk of being pedantic, the direction for brothers respecting controlling media from the stage related to the specific paragraphs in the letter concerning *Video* playback. It mentioned that 
**_"Generally, the video equipment should not be operated from the stage"._**   Note the word '**_generally_**' as well.  

Another letter mentioned visual aids and only mentioned using 'still' images, not who was controlling it. 

As I am involved in sign language we need to do what we can (we are a sign language not able to use the current supported languages) with the circumstances we have.  For example, I gave an 'away' public talk recently (using sign only) and had images as well as videos of signed scriptures (in our 'language').  I would have preferred to use OnlyM as that is the software after 2019, however we used SoundBox instead because there is an app that my son could control from his seat (we used an auxiliary classroom with ad-hoc equipment).  As he does not understand sign-language I had to 'prompt' him each time (not a biggie) but if I could have done so I likely would have... _given our circumstances_.

As a result of our use of SoundBox and the app, the overseer was so impressed he decided that they would adopt it for future use in their sign language group. 💡  

In my home congregation, for our sign-language team, we will continue to use SoundBox for the foreseeable future as each meeting we use a mobile TV and Laptop which is all controlled remotely so that the trolley with the equipment is 'hands free', and as wireless as possible... otherwise we would need a volunteer sitting in the front row with a wireless mouse trying to read what is on the laptop screen, which is not practical.  We love SoundBox and the remote apps. ❤️ 

I have also given a regular speaking talk in a congregation that had all the equipment but no one available to use it (a congregation with a number of brothers away) so I quickly installed SoundBox (I had it on my USB drive just in case) and controlled my media from my phone while giving the talk.  (I got permission from the CoBoE first). No one noticed at all as it was done super discreetly so as not to needlessly offend anyone.  📵 

However, as mentioned, respecting the sign language comments, we are not the target audience for OnlyM and I fully respect the fact that it takes much effort to develop an app.  Perhaps there simply is not enough 'call' for it.  Perhaps, once SoundBox becomes less used, brothers may assist us with a remote app... 😄

**Comment by Glipt** [comment]
Hi. Thanks for an awesome app. We have not been using Dropbox for over a year now but we really miss the remote control feature. The letter about not controlling video from the platform was referring to a time before the WT Library existed and there would have had to be a computer on the platform and the brother would be very distracted trying to use VLC or windows media player to find and present a video. Due to high crime our sound equipment and computer are locked in a very secure closet and we do not have an operator sitting at a console. Our audio is well balanced so there is never any feedback or a need to have someone constantly adjusting levels. Soundbox was awesome. We are in an isolated congregation with few brothers and at times I have been meeting chairman, given most of the talks and controlled all sound, images and video from the Soundbox Remote on an old iPhone 4. Using the Soundbox remote was so easy it was no distraction or problem doing this from the platform. I have tried using Team Viewer and Splashtop but they have proven unreliable.

Anyway our Circuit Overseer just came through and had many videos and images to use for his talks. We loaded them on to the old Soundbox and he controlled them himself during his talks. He absolutely loved it and has asked for my help getting other congregations to set this up for him. Of course there is no need to force everybody to use the remote control. If a CO or speaker has images and videos not appearing on JW Library he will always have the option to call out to the sound brother for the next image / video, etc.

For now we can just use the old Dropbox but it would be much nicer and easier to use the new OnlyM. I'm not sure if the existing Soundbox Remote app could be configured to control the OnlyM app.

Once again. Thanks for all your hard work.

**Comment by talkingdragon** [comment]
Hi. We have had the speakers ask about IMG controlling. I have had no confirmation from 2 C.O and the elders.. No such letter in Spain and all say a speaker can controll the images. It would be very useful. A option to custom keybind actions wold be rly useful so that we can then use a standard PowerPoint remote to pass images. If you can implement it. It be extremely useful.
Thks
Daniel

**Comment by AntonyCorbett** [comment]
@talkingdragon thanks for feedback. Here's the problem - JWL is the recommended meeting media player. I don't know what additional features it will provide in future releases, but it is reasonable to conclude that it will eventually support 3rd party images and videos. I am loathe to spend time and effort on implementing advanced features in OnlyM that are likely to become redundant.

**Comment by talkingdragon** [comment]
@AntonyCorbett I can understand that but then we never know how long that might take.. 🤣 Soundbox was proof of that.. U made that years ago and toke them that long to make jwapp lol

**Comment by Chipperchap** [comment]
I find the discussion on whether JWL will display third party images very interesting.  What if a brother with full naive sincerity showed an image that was actually offensive somehow?  Of course that could happen now (I stopped two speakers showing two non-jw images recently, one had an image of a collapsed exhausted runner who was very scantily clothed, and the other had a highly political image - we found alternative superior images).  The brothers would be very reluctant to have anything associated with JWL that could be inappropriate. Perhaps JWL will allow for images but only those sourced from JW.ORG or WOL... that is where I get all my images from anyway.

I have been wrong many times before though - for example I never thought that the KM would be released to the public (via JW.ORG etc.) as it was what I thought of as an 'internal journal', that is for organisation members only, and discussed some procedural matters along with ministry approaches.  Now through JWL and JW.ORG almost everything is available to any member of the public!

Sorry, I appreciate that does not add to the discussion of the feature request much.  The future for what JWL will have in even 6 or 12 months time is exciting!

**Comment by strawberry-sundae** [comment]
Hi, brothers.

Perhaps I can help by suggesting an alternative solution for speakers needing to remotely control images on screen. I haven’t had a chance to test this, but it could be worth looking into…

Apple has a free presentation app called **Keynote**. It’s available for iOS and Mac, and a web app at icloud.com works on Windows. Using a feature called *Keynote Live*, you may control your presentation from an iPhone or iPad while anyone with your permission can watch in a web browser with a link you’ll share.

See https://support.apple.com/kb/PH25953

This will at least allow you to show slides during your talk without needing the sound brother to control it from the computer.

I hope this helps you brothers with more unique circumstances. 😊

If Only M gains remote control ability to assist the relatively few of us who need it, the feature will likely be used by others who do not need it. That is what happens with technology, unfortunately. So I think it is wise of Brother Antony to avoid catering to the needs of a few with a feature that will be overused against the organisation’s guidelines.

If anyone tries my suggestion, I’d be curious to know how it goes for you. 😃

**Comment by Xeway** [comment]
Hey,

Now that we have tablets for the Zoom participants next to the orator, why not creating a website connected to OnlyM on PC? That website could be hosted on the PA computer, so that it's local and only devices connected to the wi-fi would have an access, + no need to buy external servers. If it's inconvenient to use it on the Zoom tablet, the orator could just connect to the wi-fi and go on the website, to use it on his phone.

For the technical specifications, I was thinking of building an API in the OnlyM app, with endpoints like `/display/{image_id}`, `/undisplay/{image_id}`, `/get_images`.
And aside building the website. The website would be just interacting with the API, nothing fancy. And when OnlyM starts, also start the server (maybe add an option to turn on/off the feature).
> Instead of a "forward" & "back" buttons, maybe just display on the website a similar UI to OnlyM.

What do you think?
